cmd/fleet/handleEnroll.go

```diff
@@ -293,12 +293,22 @@ func createFleetAgent(ctx context.Context, bulker bulk.Bulk, id string, agent mo
 }
 
 func generateAccessApiKey(ctx context.Context, client *elasticsearch.Client, agentId string) (*apikey.ApiKey, error) {
-	return apikey.Create(ctx, client, agentId, "", []byte(kFleetAccessRolesJSON))
+	return apikey.Create(ctx, client, agentId, "", []byte(kFleetAccessRolesJSON),
+		apikey.Metadata{
+			Application: apikey.FleetAgentApplication,
+			AgentId:     agentId,
```

Comment:
Torn if we should call it `agent_id` or `agent.id`. to be consistent with ECS. `agent.id` makes the code more complicated I guess?

Reply:
@tsg FYI, we already add the agent id. Any opinion on format?

Reply:
Ah, that's awesome, thanks for the ping. Also FYI @andrewkroh.

I'd say that we don't really need the `.` here, since we expect this to be internal. Unless it can be added without headaches.

Reply:
just had similar debate on the other PR https://github.com/elastic/kibana/pull/95935
user.id vs user_id .... was asked to change to user_id.
can do either way, just let me know

Reply:
😆  Can't see the discussion on the commit change. Was the reason there? In the end agree with @tsg it probably does not matter here, especially if in kibana we also use `user_id`. I always have an ECS preference but I probably spent too much time with ECS :-)

Reply:
it was a discussion on Slack with one of the reviewers, thus few commits there iterating on the final shape